### DISCUSSION TITLE
A new component for code block similar to streamlit's st.code()

### DISCUSSION THREAD

**Discussion by yipeng-jiang** [zauberzeug/nicegui#592] [Ideas / Feature Requests]
I am wondering if it is possible to have a similar component as the streamlit st.code(). Currently, I am uisng the ui.mardown to show code block. However, it does not highlight or change the color of the key words like **Select** for sql, or  **import** for python. Thanks.

**Comment by falkoschindler** [comment]
That's a great idea! It shouldn't be too hard. Here is how we do it for the NiceGUI website:

````py
from pygments.formatters import HtmlFormatter

ui.add_head_html(f'<style>{HtmlFormatter(nobackground=True).get_style_defs(".codehilite")}</style>')

ui.markdown('''
```python
from nicegui import ui

ui.label('Hello world')

ui.run()
```
''')
````

A `ui.code()` element would need to make sure to add the head HTML (but only once!) and wrap the `ui.markdown` element with triple backticks. Pygments supports over 500 languages, so `ui.code()` would be very flexible in this regard.

**Reply by yipeng-jiang** [reply]
Thank you for your quick answer. It will just do it.  
By the way, I just want to say how so much I appriciate you guys for creating such an excellent framework, I have been looking all kinds of framework like streamlit, dash, pynecone and even thinking about doing front end and backend separately using vue,react and flask, but the complexity of learning all this framework and connect them together just scare me out.  I have only touched the surface of the framework and I can't stop loving it already because I feel like it is best so far.

**Comment by falkoschindler** [comment]
As a first step, we decided to introduce syntax highlighting for code blocks in `ui.markdown`: #612

**Reply by yipeng-jiang** [reply]
Thank you for continuing  supporting Nicegui.

**Comment by falkoschindler** [comment]
Here is another request for something like `st.code`: #1536.

**Comment by falkoschindler** [comment]
I'll close this feature request as completed in #1653.
